metrics_utility/test/test_extra_params_validation.py

```diff
@@ -0,0 +1,98 @@
+import os
+
+import pytest
+
+from openpyxl import load_workbook
+from pandas import read_excel
+
+from metrics_utility import prepare
+from metrics_utility.test.util import run_build_int, run_gather_int
+
+
+prepare()
+
+env_vars = {
+    'METRICS_UTILITY_SHIP_PATH': './metrics_utility/test/test_data',
+    'METRICS_UTILITY_SHIP_TARGET': 'directory',
+    'METRICS_UTILITY_REPORT_TYPE': 'CCSPv2',
+    'METRICS_UTILITY_OPTIONAL_CCSP_REPORT_SHEETS': 'ccsp_summary,managed_nodes,indirectly_managed_nodes,usage_by_organizations',
+}
+file_path = './metrics_utility/test/test_data/reports/2025/02/CCSPv2-2025-02-25--2025-02-26.xlsx'
+
+
+def test_uses_since_and_until_values():
+    env_vars['METRICS_UTILITY_REPORT_TYPE'] = 'CCSPv2'
+    args = {'since': '2025-02-25', 'until': '2025-02-26', 'force': True}
+    run_build_int(env_vars, args)
+    try:
+        workbook = load_workbook(filename=file_path)
+        worksheet = read_excel(file_path, sheet_name='Usage Reporting')
+        cell_value = worksheet.iat[3, 1]
+
+        assert cell_value == '2025-02-25, 2025-02-26', f"Expected '2025-02-25, 2025-02-26' but got '{cell_value}'"
+    finally:
+        workbook.close()
+        os.remove(file_path)
+
+
+month_file_path = './metrics_utility/test/test_data/reports/2025/04/CCSPv2-2025-04.xlsx'
+
+
+def test_accepts_month_and_ignores_since_until():
+    handle_build_exception(
+        env_vars,
+        {'since': '2024-01-01', 'until': '2024-02-01', 'month': '2025-04', 'force': True},
+        'The --since and --until parameters are not allowed if the --month parameter is provided.',
+    )
+
+
+def test_renewal_guidancev2_fails_with_until_params():
+    command_args = [
+        {'args': {'until': '2025-01-01'}, 'report_type': 'RENEWAL_GUIDANCE'},
+        {'args': {'until': '2025-01-01'}, 'report_type': 'RENEWAL_GUIDANCEv2'},
+        {'args': {'until': '2025-01-01', 'since': '2024-02-01'}, 'report_type': 'RENEWAL_GUIDANCE'},
+    ]
+    for arg in command_args:
+        handle_build_exception(
+            {**env_vars, 'METRICS_UTILITY_REPORT_TYPE': arg['report_type']},
+            arg['args'],
+            'The --until parameter is not allowed when environment variable METRICS_UTILITY_REPORT_TYPE is RENEWAL_GUIDANCE or RENEWAL_GUIDANCEv2',
```

Comment:
```suggestion
            'The --until parameter is not allowed when environment variable METRICS_UTILITY_REPORT_TYPE is RENEWAL_GUIDANCE',
```

Reply:
Also .. this test didn't fail, even when the code outputting the message was changed (https://github.com/ansible/metrics-utility/pull/128/files#diff-74815671e759671fde1bdd5f0276ffd27e215e903eb3dc74b3f3dc1da16c3380R95)

So .. might be worth investigating why it didn't fail when the message doesn't match now.